Which failing test do you mean? I only saw this failing test in CI:

```
  ======================================================================
  ERROR: test_array_np_shape_dim_check (test_array.TestArray.test_array_np_shape_dim_check)
  ----------------------------------------------------------------------
  Traceback (most recent call last):
    File "D:\a\mlx\mlx\python\tests\test_array.py", line 771, in test_array_np_shape_dim_check
      mx.array(a_npy)
      ~~~~~~~~^^^^^^^
  OverflowError: Shape dimension 2147483648 is outside the supported range [-2147483648, 2147483647]. MLX currently uses 32-bit integers for shape dimensions.
  
  ----------------------------------------------------------------------
```

--- a/python/tests/test_array.py
+++ b/python/tests/test_array.py
@@ -767,10 +767,13 @@ class TestArray(mlx_tests.MLXTestCase):
 
     def test_array_np_shape_dim_check(self):
         a_npy = np.empty(2**31, dtype=np.bool_)
-        with self.assertRaises(ValueError) as e:
+        with self.assertRaises(OverflowError) as e:
             mx.array(a_npy)
         self.assertEqual(
-            str(e.exception), "Shape dimension falls outside supported `int` range."
+            str(e.exception),
+            "Shape dimension 2147483648 is outside the supported range "
+            "[-2147483648, 2147483647]. MLX currently uses 32-bit integers "
+            "for shape dimensions.",
         )
 
     def test_dtype_promotion(self):

